src/app/app.component.ts

```diff
@@ -66,6 +73,16 @@ export class AppComponent {
     });
   }
 
+  private sendPageView(e: NavigationEnd): void {
+    if (environment.production) {
+      ga('set', 'page', e.urlAfterRedirects);
```

Comment:
If the user uses an ad blocker, `ga` won't be defined. Can you add a further check?

```js
if (environment.production) {
  if (ga) {
    // ga(...)
  }
} else {
  // console.info(...)
}
```

Reply:
If you send `e.urlAfterRedirects` to Google, you will send a relative URL, such as `/private/fields/categories?lang=en`.

Instead, you need to send the full URL, which you can get with:
```js
new URL(e.urlAfterRedirects, location.origin).href
```